File: 3228-maximum-number-of-operations-to-move-ones-to-the-end/3228-maximum-number-of-operations-to-move-ones-to-the-end.py
class Solution:
    def maxOperations(self, s: str) -> int:
        n = len(s)
        ans, ones = 0, 0

        for i in range(n):
            if s[i] == '1':
                ones += 1
            elif i + 1 == n or s[i + 1] == '1':
                ans += ones

        return ans

File: 3228-maximum-number-of-operations-to-move-ones-to-the-end/3228-maximum-number-of-operations-to-move-ones-to-the-end.cpp
class Solution {
public:
    int maxOperations(string s) {
        int ans = 0, ones = 0, n = s.size();

        for (int i = 0; i < n; ++i) {
            if (s[i] == '1') ++ones;
            else if (i + 1 == n || s[i + 1] == '1') ans += ones;
        }

        return ans;
    }
};